test/unit/traversing.js

```diff
@@ -481,6 +481,19 @@ QUnit.test( "not(Selector) excludes non-element nodes (gh-2808)", function( asse
 	assert.deepEqual( mixedContents.not( "[id=a],*,[id=b]" ).get(), [], "not [id=a],*,[id=b]" );
 } );
 
+QUnit.test( "not(arraylike) passes non-element nodes (gh-3226)", function( assert ) {
+	assert.expect( 4 );
+
+	var mixedContents = jQuery( "#nonnodes" ).contents(),
+		mixedLength = mixedContents.length,
+		childElements = q( "nonnodesElement" );
+
+	assert.deepEqual( mixedContents.not( mixedContents.get() ).get(), [], "not everything" );
+	assert.deepEqual( mixedContents.not( childElements ).length, mixedLength - 1, "not childElements" );
+	assert.deepEqual( mixedContents.not( [ childElements[ 0 ].nextSibling ] ).length, mixedLength - 1, "not textnode" );
```

Comment:
Yeah, this is very fragile test, since it doesn't use generated fixtures.

Beside that, sounds like we need all node types? At least `text` and `comment`, maybe processing node too, not sure if we need to test/support deprecated nodes though.

It would be cool if it would be shown explicitly either through the code or with comment or through assertion name